core/task/class.go

```diff
@@ -101,28 +101,26 @@ func (c *Class) MarshalYAML() (interface{}, error) {
 	type _class struct {
 		Name        string                  `yaml:"name"`
 		Defaults    map[string]string       `yaml:"defaults"`
```

Comment:
Some of these can be tagged `omitempty`, including `Defaults`, `Bind`, and `Properties`.

Reply:
I did try that but it seems to have no effect. A `default: []` keeps getting marshalled regardless. Similar issues with the others. I'll go ahead an tag them though.

Reply:
For example, this is the output when tagged with `omitempty`:
```yaml
connect: []
constraints: []
defaults: null
```

I'll make a JIRA issue.